core/trino-spi/src/main/java/io/trino/spi/metrics/Histogram.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.spi.metrics;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Arrays;
+
+public class Histogram
```

Comment:
IIUC, adding airlift's `stats` to `trino-spi` seems to require adding even more dependencies to the SPI - e.g. `guava` (which is currently is used only for tests). However, after adding `guava` as a `provided` dependency [1] I got the following errors when building `trino-tpch`:
```
$ ./mvnw install -Dmaven.gitcommitid.skip=true -DskipTests -pl plugin/trino-tpch -am 
...
[ERROR] Failed to execute goal io.trino:trino-maven-plugin:10:check-spi-dependencies (default-check-spi-dependencies) on project trino-tpch: 
[ERROR] 
[ERROR] Trino plugin dependency org.hdrhistogram:HdrHistogram must not have scope 'provided'. It is not part of the SPI and will not be available at runtime.
```

Since `org.hdrhistogram` is an airlift `stats` dependency [2], I tried to add the following fixes to `trino-tpch` [3], but it seems that I didn't specified the dependency correctly :(

I would appreciate if you could please help me to find the correct way of adding the `TDigest` to `trino-spi`...

[1] https://github.com/rzeyde-varada/prestosql/commit/5b80288ec8bfc628bfc7071e0784877f6fa0c094
[2] https://github.com/airlift/airlift/blob/1e50fbd21c43ea355b7415af6a39ca7ad42c1daa/stats/pom.xml#L20
[3] https://github.com/rzeyde-varada/prestosql/commit/d1228f30554442a240517344e9cedd1051befbbe

Reply:
Thanks!
I think that I will move the implementation into a toolkit.

Reply:
Done in https://github.com/trinodb/trino/pull/6232/commits/b50d4735e4da4d736b632f8dd0e7c9bf0e50db2e.